io/module.js

```diff
@@ -93,13 +93,17 @@ const register = {
     } = options
 
     const { ext: ioSvcExt } = pParse(ioSvc)
-    const { ext: nspDirExt } = pParse(ioSvc)
+    const { ext: nspDirExt } = pParse(nspDir)
     const extList = ['.js', '.ts', '.mjs']
-    const ioSvcFull = pResolve(
-      extList.includes(ioSvcExt) ? ioSvc : ioSvc + '.js'
-    )
+    const ioSvcFull = ioSvcExt
+      ? pResolve(ioSvc)
+      : extList
+          .map((ext) => pResolve(ioSvc + ext))
+          .find((path) => existsSync(path))
```

Comment:
Ok, this should be ok. This will give additionally priority to filetype extensions. So, if someone had both an io.js and io.ts file, the io.js would be the one used.